direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.blob;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.io.ByteStreams;
+import cz.o2.proxima.annotations.Internal;
+import cz.o2.proxima.direct.bulk.AbstractBulkFileSystemAttributeWriter;
+import cz.o2.proxima.direct.bulk.FileSystem;
+import cz.o2.proxima.direct.bulk.Path;
+import cz.o2.proxima.direct.core.BulkAttributeWriter;
+import cz.o2.proxima.direct.core.CommitCallback;
+import cz.o2.proxima.direct.core.Context;
+import cz.o2.proxima.repository.EntityDescriptor;
+import cz.o2.proxima.storage.StreamElement;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/** {@link BulkAttributeWriter} for blob storages. */
+@Internal
+@Slf4j
+public abstract class BulkBlobWriter<BlobT extends BlobBase, AccessorT extends BlobStorageAccessor>
+    implements BulkAttributeWriter {
+
+  @Getter private final AccessorT accessor;
+  @Getter private final Context context;
+  private transient BulkAttributeWriter wrap;
+
+  @Getter(AccessLevel.PACKAGE)
+  private transient boolean initialized;
+
+  public BulkBlobWriter(AccessorT accessor, Context context) {
+    this.accessor = accessor;
+    this.context = context;
+    init();
```

Comment:
I thought that this causes trouble during deserializaation. Should we remove this?

Reply:
This happens in constructor only. But it might be unnecessary, so I'll try to remove that.

Reply:
removed